conans/client/cmd/new.py

```diff
@@ -289,6 +316,17 @@ def cmd_new(ref, header=False, pure_c=False, test=False, exports_sources=False,
         t = Template(conanfile_template)
         replaced = t.render(name=name, version=version, package_name=package_name)
         files = {"conanfile.py": replaced}
+    elif template_dir:
+        if not os.path.isabs(template_dir):
+            template_dir = os.path.join(cache.cache_folder, "templates", template_dir)
```

Comment:
I'd really want to think about this. We already have `templates/outputs` for the table and graph templates, I would vote to add something like `commands/new` to this path and maybe the path from the command line (if relative) can start to look for the files inside this directory, wdyt? 

This is only a suggestion, I want to see other alternatives

Reply:
> I would vote to add something like commands/new to this path and maybe the path from the command line (if relative) can start to look for the files inside this directory, wdyt?

Sorry, I've failed to parse this sentence, could you rephrase that plz? What do you mean by "add something like commands/new to this path"?

Reply:
Sorry, I thought mine was a PR already released, but it isn't. We are introducing a way for the user to override the templates for the search table output and the graph info. Here you can see the PR to the docs: https://github.com/conan-io/docs/pull/1739.

Those templates are expected to be inside the directory `<cache>/templates/output/`. I expect more templates to arrive (like this PR you are suggesting here), so we need to organize that `templates` folder. My suggestion is to store templates for the `conan new` command inside a `<cache>/templates/commands/new/` directory.

The user, then, will use this new feature like:

* with an absolute path, everything clear:
   ```
   conan new --template /abs/path/to/folder
   ```
* with a relative one:
   ```
   conan new --template path/to/folder
   ```

   Now Conan can look first inside the working directory `$(pwd)/path/to/folder` and then inside the cache folder `<cache>/templates/commands/new/path/to/folder`.

   If using just a single name, it would be easier to understand, for example:

   ```
   conan new --template header_only
   ```
   
   Will get templates from `<cache>/templates/commands/new/header_only/`


I'm open to other alternatives, this is just a suggestion.

Reply:
Seems reasonable. Though I'm not sure that `commands` subdir is needed (i.e. `cmd_new` would be fine), since AFAIK there are no other conan commands that can (or could) use templates.

Btw, what about plain `.py` templates? Should they go into that new folder as well? Or maybe they should be deprecated altogether in favor of directory based templates?

Reply:
We need to keep supporting files as they were working before, but if something is better and breaking we can consider it as we are already designing the new Conan 2.0.

Reply:
fixed: 
- used `cmd_new` dir name
- added support for new subfolder in the old `.py` template case